sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/RowEncoder.scala

```diff
@@ -187,7 +187,7 @@ object RowEncoder {
         val convertedField = if (field.nullable) {
           If(
             Invoke(inputObject, "isNullAt", BooleanType, Literal(index) :: Nil),
-            Literal.create(null, field.dataType),
+            Literal.create(null, fieldValue.dataType),
```

Comment:
can we add a comment to explain it? `field.dataType` can be different from `fieldValue.dataType`, because we strip UDT

Reply:
Added a comment.